src/async_impl/client.rs

```diff
@@ -2336,6 +2330,18 @@ impl Default for Client {
     }
 }
 
+#[cfg(feature = "__rustls")]
+fn default_rustls_crypto_provider() -> Arc<rustls::crypto::CryptoProvider> {
+    #[cfg(not(any(feature = "__rustls-ring", feature = "__rustls-aws-lc-rs")))]
+    panic!("No provider set");
```

Comment:
Is it worth having this? If a downstream is not setting any `__*` options this doesn't happen, right?

Reply:
I think it can still happen, if a user enabled a `*-no-provider` feature and didn't "install" a default using rustls directly, right?

Reply:
Fair!